src/python/tests/gff3/test_simplifier.py

```diff
@@ -0,0 +1,365 @@
+    with expectation:
+        new_feat = check_one_feature(input_gff, output_gff, "clean_gene")
+        if new_feat:
+            assert_files(output_gff, Path(data_dir / expected_gff))
+
+
+@pytest.mark.parametrize(
+    "in_gff, expected_gff, expectation",
+    [
+        param("ok_gene.gff", "ok_gene.gff", does_not_raise(), id="ok gene"),
+        param("gene_ignored.gff", None, does_not_raise(), id="gene ignored"),
+        param("mobile_te.gff", "mobile_te.gff", does_not_raise(), id="TE"),
+        param("ok_protein_coding_gene.gff", "ok_gene.gff", does_not_raise(), id="ok protein_coding_gene"),
+    ],
+)
+def test_simpler_gff3_feature(
+    data_dir: Path,
+    tmp_dir: Path,
+    assert_files: Callable,
+    in_gff: PathLike,
+    expected_gff: Optional[PathLike],
+    expectation: ContextManager,
+) -> None:
+    """Test simplifying one gene (from a GFF3 file)."""
+    input_gff = data_dir / in_gff
+    output_gff = tmp_dir / in_gff
+    with expectation:
+        new_feat = check_one_feature(input_gff, output_gff, "simpler_gff3_feature")
+        if expected_gff is None:
+            assert new_feat is None
+        else:
+            assert_files(output_gff, Path(data_dir / expected_gff))
+
+
+@pytest.mark.parametrize(
+    "in_gff, expected_gff, expectation",
+    [
+        param("ok_gene.gff", "ok_gene.gff", does_not_raise(), id="ok gene"),
+        param("bad_gene_type.gff", "", raises(GFFParserError), id="Unsupported gene type"),
+        param("bad_tr_type.gff", "", raises(GFFParserError), id="Unsupported transcript type"),
+        param("bad_subtr_type.gff", "", raises(GFFParserError), id="Unsupported subtranscript type"),
+    ],
+)
+def test_simpler_gff3(
+    data_dir: Path,
+    tmp_dir: Path,
+    assert_files: Callable,
+    in_gff: PathLike,
+    expected_gff: PathLike,
+    expectation: ContextManager,
+) -> None:
+    """Test simplifying genes from GFF3 files."""
+    input_gff = data_dir / in_gff
+    output_gff = tmp_dir / in_gff
+    with expectation:
+        simp = GFFSimplifier()
+        simp.simpler_gff3(input_gff)
+        simp.records.to_gff(output_gff)
+        assert_files(output_gff, data_dir / expected_gff)
+
+
+@pytest.mark.parametrize(
+    "in_gff, expected_gff, skip_unrecognized, expectation",
+    [
+        param("bad_gene_type.gff", "", None, raises(GFFParserError), id="Unset skip unrecognized, fail"),
+        param(
+            "bad_gene_type.gff",
+            "bad_gene_type_skipped.gff",
+            True,
+            does_not_raise(),
+            id="True skip unrecognized, no fail",
+        ),
+        param("bad_gene_type.gff", "", False, raises(GFFParserError), id="bad type, fail"),
+        param("ok_gene.gff", "ok_gene.gff", False, does_not_raise(), id="ok type, fail"),
+    ],
+)
+def test_simpler_gff3_skip(
+    data_dir: Path,
+    tmp_dir: Path,
+    assert_files: Callable,
+    in_gff: PathLike,
+    expected_gff: PathLike,
+    skip_unrecognized: Optional[bool],
```

Comment:
```suggestion
    skip_unrecognized: bool,
```
Signature changed before.